src/components/Header2.tsx

```diff
@@ -0,0 +1,104 @@
+import { useState } from 'react';
+import { GrClose } from 'react-icons/gr';
+import { GiHamburgerMenu } from 'react-icons/gi';
+import { Link } from 'react-router-dom';
+import { HiOutlineUserAdd } from 'react-icons/hi';
+import { IoIosLogIn, IoIosLogOut } from 'react-icons/io';
+import { CgProfile } from 'react-icons/cg';
+import { useAuth } from '../hooks';
+import { toast } from 'react-toastify';
+
+function Links() {
+  const { authenticated: isLoggedIn, setAuthenticated: setIsLoggedIn } =
+    useAuth();
+
+  const handleSignOut = () => {
+    setIsLoggedIn(false);
+  };
+
+  return (
+    <>
+      <Link to="/" className="hover:text-moonNeutral-600">
+        Home
+      </Link>
+      <Link to="./catalog" className="hover:text-gray-500">
+        Catalog
+      </Link>
+      <Link to="./about" className="hover:text-moonNeutral-600">
+        About
+      </Link>
+      <Link to="./contact" className="hover:text-moonNeutral-600">
+        Contact
+      </Link>
+      {!isLoggedIn && (
+        <Link to="./signup" title="Registration">
+          {' '}
+          <HiOutlineUserAdd className="text-2xl hover:text-moonNeutral-600" />{' '}
+        </Link>
+      )}
+      {!isLoggedIn && (
+        <Link to="./signin" title="Login">
+          {' '}
+          <IoIosLogIn className="text-2xl hover:text-moonNeutral-600" />{' '}
+        </Link>
+      )}
+      {isLoggedIn && (
+        <Link to="./profile" title="My Profile">
+          {' '}
+          <CgProfile className="text-2xl hover:text-moonNeutral-600" />{' '}
+        </Link>
+      )}
+      {isLoggedIn && (
+        <Link
+          to="./signin"
+          title="Logout"
+          onClick={() => {
+            handleSignOut();
+            toast.success('Success Logout!');
+          }}
+        >
+          {' '}
+          <IoIosLogOut className="text-2xl hover:text-moonNeutral-600" />{' '}
+        </Link>
+      )}
+    </>
+  );
+}
+
+function Logo() {
+  return (
+    <Link
+      to="/"
+      className="flex items-center h-10 px-10 font-bold uppercase italic text-white hover:opacity-90"
+    >
+      <img
+        className="h-8 w-auto"
+        src="/png/Random20Logo.png"
+        alt="Random Twenty Logo"
+      />
+    </Link>
+  );
+}
+function Header2() {
+  const [showMenu, setShowMenu] = useState(false);
+
+  return (
```

Comment:
also you can create common styles variable for your css styyles 

  const commonLinkClass = 'hover:text-moonNeutral-600';
  
  and can optimazing your JSX code inside return 
  ```

   <>
      <Link to="/" className={commonLinkClass}>
        Home
      </Link>
      <Link to="./catalog" className="hover:text-gray-500">
        Catalog
      </Link>
      <Link to="./about" className={commonLinkClass}>
        About
      </Link>
      <Link to="./contact" className={commonLinkClass}>
        Contact
      </Link>
      {!isLoggedIn ? authLinks : profileLinks}
    </>
```